dotnet-authserver/src/TeacherIdentity.AuthServer/Jobs/RefreshEstablishmentDomainsJob.cs

```diff
@@ -0,0 +1,105 @@
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using Microsoft.EntityFrameworkCore;
+using TeacherIdentity.AuthServer;
+using TeacherIdentity.AuthServer.Models;
+using TeacherIdentity.AuthServer.Services.Establishment;
+
+namespace TeacherIdentity.AuthServer.Jobs;
+
+public class RefreshEstablishmentDomainsJob
+{
+    private const string ExtractDomainRegex = "^(?:https?:\\/\\/)?(?:[^@\\/\\n]+@)?(?:www\\.)?([^:\\/\\n]+)";
```

Comment:
May a little more efficient to stash a `Regex` instance here and use source generator e.g.
```
[GeneratedRegex("^(?:https?:\\/\\/)?(?:[^@\\/\\n]+@)?(?:www\\.)?([^:\\/\\n]+)")]
private static partial Regex MatchDomainRegex();
```

Reply:
Nice! - done